pkg/api/vlabs/validate.go

```diff
@@ -601,6 +601,16 @@ func (a *Properties) validateAddons() error {
 				if to.Bool(addon.Enabled) && a.HasCoreOS() {
 					return errors.New("flexvolume add-ons not currently supported on coreos distro. Please use Ubuntu")
 				}
+			case "kube-dns":
+				if to.Bool(addon.Enabled) {
+					maxVersion, err := semver.Make("1.12.0-alpha.1")
```

Comment:
@CecileRobertMichon To be extra paranoid I changed this to `1.12.0-alpha.1` as it's the earliest 1.12 version ever supported by AKS Engine